xmrstak/misc/executor.cpp

```diff
@@ -997,6 +1001,52 @@ void executor::print_report(ex_event_name ev)
 	printer::inst()->print_str(out.c_str());
 }
 
+void executor::shutdown()
+{
+	// if shutdown is called the first time
```

Comment:
You are calling this from different threads. bQuit needs to be atomic. pvThreads is not safe. Use signals instead and wait conditions instead of calls and busy waiting.